### DISCUSSION TITLE
MUD <-> Discord bridge

### DISCUSSION THREAD

**Discussion by november-kilo** [dworkin/dgd#85] [General]
I wrote a Discord bot to bridge our MUD and Discord. Going from the MUD to Discord seems straightforward, but the reverse is not immediately apparent to me. The latter isn't necessary because we want the MUD to post to Discord, so I have yet to think about it, but that might change. I assume a `call_out()` or continuation into a queue, but I haven't tried anything. In the meantime, I wanted to ask for thoughts in general or about handling `lpc_ext_read()` when it has nothing to read in particular. 

I wrote the bot using concord: https://github.com/cogmasters/concord

The bot and the kfun extensions are here: https://github.com/november-kilo/cloud-server/commit/6e65d8aeb28f9abb10155fb1a4acd58c47bcf00f

**Comment by dworkin** [comment]
`lpc_ext_spawn()` is not intended to execute a program that can be communicated with directly from LPC, and `lpc_ext_read` and `lpc_ext_write()` should not be called from a kfun because atomic rollback would not be possible.

It would be better to write the Discord bot as an external program that connects to the MUD on a socket, so that input arrives as a proper event through `receive_message`.  Or else, write a bot completely in LPC.

**Comment by november-kilo** [comment]
Thanks. I scrapped the kfuns and connected the bot on a socket.
